tests/usage/test_usage_lib.py

```
@@ -0,0 +1,214 @@
+        mock_config.architectures = []
+
+        message = UsageMessage()
+        model_info = message._extract_model_info(mock_config)
+
+        assert model_info["architecture"] == "unknown"
+
+
+class TestReportUsage:
+    """Test report_usage function."""
+
+    def test_report_usage_respects_opt_out(self, monkeypatch, tmp_path):
```

Ruff (0.14.14)

[warning] Unused method argument: `tmp_path`

(ARG002)